pkg/splunk/enterprise/indexercluster.go

```diff
@@ -414,6 +424,40 @@ func (mgr *indexerClusterPodManager) getClusterMasterClient() *splclient.SplunkC
 	return mgr.newSplunkClient(fmt.Sprintf("https://%s:8089", fqdnName), "admin", adminPwd)
 }
 
+// getSiteRepFactorOriginCount gets the origin count of the site_replication_factor
+func getSiteRepFactorOriginCount(siteRepFactor string) int32 {
+	re := regexp.MustCompile(".*origin:(?P<rf>.*),.*")
+	match := re.FindStringSubmatch(siteRepFactor)
+	siteRF, _ := strconv.Atoi(match[1])
+	return int32(siteRF)
+}
+
+// verifyRFPeers verifies the number of peers specified in the replicas section
+// of IndexerClsuster CR. If it is less than RF, than we se it to RF.
+func (mgr *indexerClusterPodManager) verifyRFPeers(c splcommon.ControllerClient) error {
+	if mgr.c == nil {
```

Comment:
The assignment of client is done in **IndexerClusterPodManager::Update** as well. Can we move the assignment to this line as: 
`mgr := indexerClusterPodManager{log: scopedLog, cr: cr, secrets: namespaceScopedSecret, newSplunkClient: splclient.NewSplunkClient, c:client}`

Reply:
I could do that but since then the UT will fail because of the reason I gave in the below comment. So leaving it like this for now. But good suggestion though. Might have to think of some other way for this to work with UT as well.

Reply:
Agreed